pandas/core/groupby.py

```diff
@@ -2324,7 +2324,7 @@ def _cython_operation(self, kind, values, how, axis, min_count=-1):
             result = self._transform(
                 result, values, labels, func, is_numeric, is_datetimelike)
 
-        if is_integer_dtype(result):
+        if is_integer_dtype(result) and not is_datetimelike:
             mask = result == iNaT
```

Comment:
I think you can then remove this masking bizness (inside the is_integer_dtype)

Reply:
this might also fix #16674, can you add a test and see for that (and if so add to the whatsnew).

Reply:
It doesn't look like I can (easily) do this. If `resample` introduces missing values to an integer series, those get recorded as iNaT.  Since the user is expecting nan for missing values, we have to recast the series to float and explicitly convert the iNaT values.

#16674 is about the case where iNaT was in the input dataframe as a legitimate integer value. At this point in the code, there is no way to disambiguate between that and true missing values.